lib/lib_rover/include/actuators/motor_drivers/talon_srx.hpp

```diff
@@ -23,32 +22,47 @@ class TalonSrx : public MotorDriver
     static constexpr uint32_t MS_TO_DUTY(uint32_t freq_, float microSeconds_)
     {
         float period = 1.0f / static_cast<float>(freq_);
-        return static_cast<uint32_t>(round(((microSeconds_ / 1'000'000.f) / period) * 1048576.0f));
+#if defined(ARDUINO_ESP32S2_DEV)
+        return static_cast<uint32_t>(round(((microSeconds_ / 1'000'000.f) / period) * ((float)(1u << 20)));
+#elif defined(ARDUINO_ESP32S3_DEV)
+        return static_cast<uint32_t>(round(((microSeconds_ / 1'000'000.f) / period) * ((float)(1u << 14))));
+#endif
     }
 
 public:
     // Try using the same timer for all same frequency signal. Don't use the same channel
-    TalonSrx(gpio_num_t pinPWM_, ledc_timer_t timerNumber_ = LEDC_TIMER_0, ledc_channel_t channelNumber_ = LEDC_CHANNEL_0)
+    TalonSrx(gpio_num_t pinPWM_,
+             ledc_timer_t timerNumber_ = LEDC_TIMER_0,
+             ledc_channel_t channelNumber_ = LEDC_CHANNEL_0,
+             float signalFrequencyHz_ = 50.0f)
+        : MotorDriver(eBrakeMode::NONE)
     {
+#error The TalonSRX class has been updated and cleaned blindly (no available working board to test on): please take the time to test it before using
```

Comment:
Shoulldnt this be placed in the header to avoid running any  useless code. Run default error.

Reply:
"#error" is a preprocessor instruction which will fail compilation of the code with a msg if this header file is included in any code. It will need to be removed when someone use it in the future.